src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -502,42 +492,52 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
 
-                    if (returnFullPath && (rbSymlink.Flags & Interop.Kernel32.SYMLINK_FLAG_RELATIVE) != 0)
+                    bool isRelative = (rbSymlink.Flags & Interop.Kernel32.SYMLINK_FLAG_RELATIVE) != 0;
+                    if (!isRelative)
                     {
-                        // Target path is relative and is for ResolveLinkTarget(), we need to append the link directory.
-                        return Path.Join(Path.GetDirectoryName(linkPath.AsSpan()), targetPath);
-                    }
+                        // Absolute target is in NT format and we need to clean it up before return it to the user.
+                        if (targetPath.StartsWith(PathInternal.UncNTPathPrefix.AsSpan()))
+                        {
+                            // We need to prepend the Win32 equivalent of UNC NT prefix.
+                            return Path.Join(PathInternal.UncPathPrefix.AsSpan(), targetPath.Slice(PathInternal.UncNTPathPrefix.Length)).ToString();
+                        }
 
-                    return targetPath.ToString();
+                        Debug.Assert(targetPath.StartsWith(PathInternal.NTPathPrefix.AsSpan()));
+                        return targetPath.Slice(PathInternal.NTPathPrefix.Length).ToString();
+                    }
+                    else if (returnFullPath)
+                    {
+                        return Path.Join(Path.GetDirectoryName(linkPath.AsSpan()), targetPath).ToString();
+                    }
+                    else
+                    {
+                        return targetPath.ToString();
+                    }
                 }
                 else if (rbSymlink.ReparseTag == Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_MOUNT_POINT)
                 {
                     success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.MountPointReparseBuffer rbMountPoint);
                     Debug.Assert(success);
 
-                    int printNameOffset = sizeof(Interop.Kernel32.MountPointReparseBuffer) + rbMountPoint.PrintNameOffset;
-                    int printNameLength = rbMountPoint.PrintNameLength;
+                    int offset = sizeof(Interop.Kernel32.MountPointReparseBuffer) + rbMountPoint.SubstituteNameOffset;
+                    int length = rbMountPoint.SubstituteNameLength;
 
-                    Span<char> targetPath = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(printNameOffset, printNameLength));
+                    Span<char> targetPath = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(offset, length));
 
-                    // Unlike symlinks, mount point paths cannot be relative
+                    // Unlike symbolic links, mount point paths cannot be relative.
                     Debug.Assert(!PathInternal.IsPartiallyQualified(targetPath));
-                    return targetPath.ToString();
+                    Debug.Assert(targetPath.StartsWith(PathInternal.NTPathPrefix.AsSpan()));
```

Comment:
Should we also have an assert that checks that the path does _not_ begin with the UNC prefix?

```cs
Debug.Assert(!targetPath.StartsWith(Path.Internal.UncNTPathPrefix.AsSpan()));
```

Reply:
Good point, from what I've read, I haven't seen an instance in where a mount point can point to a remote, junctions explicitly state that [remote locations are not supported](https://docs.microsoft.com/en-us/sysinternals/downloads/junction).